MergerService/src/run.cs

```diff
@@ -51,153 +59,259 @@ private List<IData> BuildDataList(Source[] paths, int batchSize)
             }
         }
 
+        private List<KeyValuePair<string, string>> BuildTypeList()
+        {
+            var taskTypes = this._configurationManager.GetChildren("TASK", "types");
+
+            List<KeyValuePair<string, string>> values = new List<KeyValuePair<string, string>>();
+
+            foreach (var pair in taskTypes)
+            {
+                var jobType = pair.GetValue<string>("JobType");
+                var taskType = pair.GetValue<string>("taskType");
+                values.Add(new KeyValuePair<string, string>(jobType, taskType));
+            }
+
+            return values;
+        }
+
         public void Start()
         {
-            Stopwatch stopWatch = new Stopwatch();
-            TimeSpan ts;
+            var pollingTime = this._configurationManager.GetConfiguration<int>("TASK", "pollingTime");
+
+            var taskTypes = BuildTypeList();
+            ITaskUtils taskUtils = new TaskUtils(this._configurationManager, this._requestUtils, this._taskUtilsLogger, this._activitySource);
+
             this._logger.LogInformation("starting task polling loop");
             while (true)
             {
                 MergeTask? task = null;
+                bool activatedAny = false;
 
-                try
-                {
-                    task = MergeTask.GetTask(this._mergeTaskLogger);
-                }
-                catch (Exception e)
+                foreach (var item in taskTypes)
                 {
-                    this._logger.LogError($"Error in MergerService run - get task: {e.Message}");
+                    string jobType = item.Key;
+                    string taskType = item.Value;
+
+                    try
+                    {
+                        task = taskUtils.GetTask(jobType, taskType);
+                    }
+                    catch (Exception e)
+                    {
+                        this._logger.LogError($"Error in MergerService start - get task: {e.Message}");
+                    }
+
+                    // Guard clause in case there are no batches or sources
+                    if (task == null)
+                    {
+                        continue;
+                    }
+
+                    task.Status = Status.IN_PROGRESS;
+                    RunTask(task, taskUtils);
+                    activatedAny = true;
+
+                    try
+                    {
+                        task.MarkCompleted();
+                        UpdateParameters updateParameters = task.GetUpdateParameters();
+                        taskUtils.UpdateTask(task.JobId, task.Id, updateParameters);
+                        taskUtils.UpdateCompletion(task.JobId, task.Id);
+                    }
+                    catch (Exception e)
+                    {
+                        this._logger.LogError($"Error in MergerService start - update task: {e.Message}");
```

Comment:
should be marked as failed or pending (depending on error type and attempts count)

Reply:
Added logic.